save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/usersettings/SettingsView.kt

```diff
@@ -0,0 +1,180 @@
+/**
+ * A view with settings user
+ */
+
+@file:Suppress("FILE_NAME_MATCH_CLASS")
+
+package com.saveourtool.save.frontend.components.views.usersettings
+
+import com.saveourtool.save.frontend.components.inputform.InputTypes
+import com.saveourtool.save.frontend.components.views.index.*
+import com.saveourtool.save.frontend.components.views.usersettings.right.FieldsStateSetter
+import com.saveourtool.save.frontend.externals.fontawesome.*
+import com.saveourtool.save.frontend.utils.*
+import com.saveourtool.save.info.UserInfo
+import com.saveourtool.save.validation.FrontendRoutes
+
+import js.core.jso
+import org.w3c.fetch.Headers
+import react.*
+import react.dom.html.ReactHTML.div
+import react.dom.html.ReactHTML.input
+import react.dom.html.ReactHTML.main
+import web.cssom.*
+import web.html.InputType
+
+import kotlinx.browser.window
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+
+val cardHeight: CSSProperties = jso {
+    height = 50.rem
+}
+
+val userSettingsView: FC<SettingsProps> = FC { props ->
+    useBackground(Style.SAVE_LIGHT)
+    main {
+        className = ClassName("main-content")
+        div {
+            className = ClassName("page-header align-items-start min-vh-100")
+            div {
+                className = ClassName("row justify-content-center mt-3")
+                div {
+                    className = ClassName("col-2")
+                    leftColumn { this.userInfo = props.userInfo }
+                }
+                div {
+                    className = ClassName("col-7")
+                    props.userInfo?.let {
+                        rightColumn {
+                            this.userInfo = props.userInfo
+                            this.type = props.type
+                        }
+                    } ?: main {
+                        // FixMe: some light 404
+                    }
+                }
+            }
+        }
+    }
+}
+
+/**
+ * `Props` retrieved from router
+ */
+@Suppress("MISSING_KDOC_CLASS_ELEMENTS")
+external interface SettingsProps : PropsWithChildren {
+    /**
+     * Currently logged in user or null
+     */
+    var userInfo: UserInfo?
+
+    /**
+     * just a flag for a factory
+     */
+    var type: FrontendRoutes
+}
+
+/**
+ * Drawing an input for profile settings (one-liner)
+ *
+ * @param previousValue
+ * @param inputType
+ * @param fieldsMap
+ * @param fieldsValidationMap
+ * @param setFieldsMap
+ * @param placeholderText
+ */
+@Suppress("TOO_MANY_PARAMETERS", "LongParameterList")
+fun ChildrenBuilder.inputForm(
+    previousValue: String?,
+    inputType: InputTypes,
+    fieldsMap: MutableMap<InputTypes, String?>,
+    fieldsValidationMap: MutableMap<InputTypes, String?>,
```

Comment:
The idea is that we need to add elements to the map. 
We collect all values from several inputs - it could probably be a typed data class, but in this particular case it is much easier and beautiful to have a map and get `value or null`.

Reply:
The suggestion is to use immutable `Map`, as you might want to change `MutableMap` without using `StateSetter`

Reply:
> The suggestion is to use immutable `Map`, as you might want to change `MutableMap` without using `StateSetter`

This is exactly what I do, I change it without StateSetter... Got your point.

Reply:
It was completely reworked with a data class @sanyavertolet. Thank you for a good point